middie.js

```diff
@@ -52,17 +54,19 @@ function middie (complete) {
       const url = that.url
       const i = that.i++
 
-      if (err || functions.length === i) {
+      if (err || urls.length === i) {
         complete(err, req, res)
         that.req = null
         that.res = null
         that.i = 0
         pool.release(that)
       } else {
-        if (!urls[i]) {
-          functions[i](req, res, that.done)
-        } else if (urls[i] && (urls[i] === url || urls[i].indexOf(url) > -1)) {
-          functions[i](req, res, that.done)
+        if (!urls[i].path) {
+          urls[i].fn(req, res, that.done)
+        } else if (urls[i].wildcard && pathMatchWildcard(url, urls[i].path)) {
+          urls[i].fn(req, res, that.done)
+        } else if (urls[i].path === url || (typeof urls[i].path !== 'string' && urls[i].path.indexOf(url) > -1)) {
+          urls[i].fn(req, res, that.done)
```

Comment:
About naming, feel to change whatever seems unclear to you. I always have hard times giving names.

> urls[i].fn() will set this to urls[i], maybe this was happening before as well, but I think we should change it.

Not sure wich context we want to assign to that function.